ctapipe/containers.py

```diff
@@ -165,6 +165,67 @@ class EventType(enum.Enum):
     UNKNOWN = 255
 
 
+class PixelStatus(enum.IntFlag):
+    """
+    Pixel status information
+
+    See DL0 Data Model specification:
+    https://redmine.cta-observatory.org/dmsf/files/17552/view
+    """
+
+    DVR_STORED_AS_SIGNAL = enum.auto()
+    DVR_STORED_NO_SIGNAL = enum.auto()
+    HIGH_GAIN_STORED = enum.auto()
+    LOW_GAIN_STORED = enum.auto()
+    SATURATED = enum.auto()
+    PIXEL_TRIGGER_0 = enum.auto()
+    PIXEL_TRIGGER_1 = enum.auto()
+    PIXEL_TRIGGER_2 = enum.auto()
+
+    #: DVR status uses two bits
+    #: 0 = not stored, 1 = identified as signal, 2 = stored, not identified as signal
+    DVR_STATUS = DVR_STORED_AS_SIGNAL | DVR_STORED_NO_SIGNAL
+
+    #: Pixel trigger information, TBD
+    PIXEL_TRIGGER = PIXEL_TRIGGER_0 | PIXEL_TRIGGER_1 | PIXEL_TRIGGER_2
+
+    @staticmethod
+    def get_dvr_status(pixel_status):
+        """
+        Return only the bits corresponding to the DVR_STATUS
+
+        Returns
+        -------
+        dvr_status: int or array[uint8]
+            0 = pixel not stored
+            1 = pixel was identified as signal pixel and stored
+            2 = pixel was stored, but not identified as signal
+        """
+        return pixel_status & PixelStatus.DVR_STATUS
+
+    @staticmethod
+    def get_channel_info(pixel_status):
+        """
+        Return only the bits corresponding to the channel info (high/low gain stored)
+
+        Returns
+        -------
+        channel_info: int or array[uint8]
+            0 = pixel broken / disabled
+            1 = only high gain read out
+            2 = only low gain read out
+            3 = both gains read out
+        """
+        gain_bits = PixelStatus.HIGH_GAIN_STORED | PixelStatus.LOW_GAIN_STORED
+        return (pixel_status & gain_bits) >> 2
+
+    @staticmethod
+    def is_broken(pixel_status):
```

Comment:
Since people often complain about the term "broken" here, maybe rename this `is_excluded()` to be more clear that the pixel may be excluded for reasons other than being physically broken (e.g. disabled, noisy, etc).

Reply:
Then I'd say disabled or missing is the correct term, as at R1 / DL0 it means whether the pixel was read out at all.

Reply:
Actually, in the R1 model, all pixels are always read out, but may have bogus values in them (the readout is fixed-length).   In the discussions about the interface, it seems this field may also be used for noisy pixels (e.g. those that are read out, but maybe not good for analysis due to a star or something), so _excluded_ is perhaps the better term.  It's different from the DVR flag, which says if this data are stored or not at the DL0 level. 

So this flag means one of two things:
* the pixel is physically broken/missing (i.e. it has an undefined signal in the readout that may be 0 or a junk number)
* the pixel has a high pedestal variance, meaning that it has a hardware fault, or has excess unexpected external light hitting it for some reason (bright star where the HV wasn't adjusted, etc)
* the pixel has a high _predicted_ light level and the HV was turned off or reduced due to an expected star. In that case, there may be some signal, but it is not properly calibrated and should be excluded.

Reply:
Note that you would get these values even if the pixel is not stored due to DVR (the pixels_status is always reported)

Reply:
Invalid is fine, and also nice as we use that term for the parameters as well.